### ISSUE TITLE
Sync Mod issue

### ISSUE THREAD

**Issue by Diociae** [Darkosto/SkyFactory4Issues#880]
## Bug Report

Sync didn't work in the Hardcore Map Variant, I died to test after filling both the Constructor and Storage with a shell, but I am forced to either Spectate or Delete the world.

## Expected Behaviour

I should have the option to respawn into either the Constructor or Storage 's Shells

## Steps to Reproduce (for bugs)

1.Fill both the Constructor and Storage with a Shell and Die
2. Don't get to respawn

## Logs

* Client/Server Log: https://gist.github.com/053003dab888dc08f3375aa01a54094b.git
* Crash Log:

## World Information

* Preset: Hardcore

* Prestige: Yes

* Modpack Version world created in: 4.0.5    / Latest

## Client Information

* Modpack Version: 4.0.5  / Latest
* Java Version: Not Sure
* Launcher Used: Twitch
* Memory Allocated: 12288 MB
* Server/LAN/Single Player: Single Player
* Optifine Installed: Don't think so
* Shaders Enabled: No

**Comment by sk2048** [comment]
I don't think you can ever respawn into the constructor, when you respawn you are taken to a shell in a shell storage that is receiving redstone power. Was your storage powered?

**Comment by CleeKru** [comment]
Had the same issue. Checked redstone and shell tried setting it to home shell all resulted in death.

**Comment by griseouslight** [comment]
I can confirm that mine was powered and I did not have the option to respawn. 
![2019-05-20_15 45 51](https://user-images.githubusercontent.com/9774133/58047606-85744f80-7b16-11e9-9b7c-592af44bf888.png)
Taken in spectator mode.

**Comment by griseouslight** [comment]
A bit of an update, this only happens when I use MultiMC instead of the Twitch launcher. I'm not sure what the difference is here as I've installed the same version of forge and used a symlink to the twitch install directory.

I will personally just use twitch for now.

**Comment by CleeKru** [comment]
It happens for me in both launchers. So that's no fix for me.

**Comment by Xetaxheb** [comment]
Having this issue as well.

**Comment by Xetaxheb** [comment]
Getting a lot of these errors in log as well:

[12:23:16] [Client thread/INFO] [STDERR]: [me.ichun.mods.sync.common.tileentity.TileEntityDualVertical:func_145839_a:409]: Read playerUUID:null for name 

and 

[12:25:07] [Client thread/INFO] [STDERR]: [me.ichun.mods.sync.common.tileentity.TileEntityDualVertical:func_145839_a:409]: Read playerUUID:  (my uuid+username)

**Comment by Darkosto** [comment]
We've tested this multiple times with multiple people and it worked properly. The advancements were added specifically to address player's not knowing how to use the Sync mod since its redesign. Can you confirm you followed advancements when trying to setup your shell?

**Comment by vonriel** [comment]
I had the same issue, and recorded a video of it happening to me. [You can watch it here.](https://youtu.be/MwnMYKb3qgU)

This was originally in a single player world, where I died after hopping off. At that point, I spectated, opened it to LAN and enabled cheats, then set myself back to Survival mode so I could record the above video.

Other people have noted on Sync's issue page that void damage follows you to your next shell and then kills it, too, chaining apparently infinitely and killing them all.
